### PR TITLE
126 list item

### PR DESCRIPTION
**Pull request by TristanHoladay**

### CODE REVIEW

**Comment by TristanHoladay** [comment]
Thinking about the merit of using event dispatchers for ListItem responding to adornment (child) events. Though it does look like, according to Rich Harris, allowing for default event bubbling is an acceptable approach. 

https://stackoverflow.com/questions/56857853/is-it-possible-to-make-a-child-component-inside-a-slot-to-send-data-to-his-paren

**Review by Madeline-UX — CHANGES_REQUESTED** [review]
- Would like to meet to align language around list vs list item so we can make sure the design file is using the same vocabulary. 
-  icons and the checkboxes need to respond to dark theme
Screenshot 2023-03-01 at 4 48 15 PM
- Nested list items show a nested item that does not align with the other list items
Screenshot 2023-03-01 at 4 47 39 PM
- Would like to see more examples of the variants.

**Review by Madeline-UX — CHANGES_REQUESTED** [review]
- [x] Move List component documentation to the bottom of page

- [ ] Change page name to List item

- [ ] Change "The ListItem is the true workhorse of List. It wraps your text as well as adornments, like icons, icon buttons, images, checkboxes, etc..." to "The ListItem is the base component of any List. A list item is used to wrap text and adornments into a single component. List item adornment icons, icon buttons, images, checkboxes, and avatars."

- [ ] Update List item flow/examples to this outline

- List Item
   - Props
      - Text
      - Variant
      - Secondary text
      - Selected
      - Disabled
      - Divider
      - Disabled Gutters
- Variants:
   - Simple
   - Icon
   - Avatar
- Slots
    - Describe Slots
   - Examples
      - Leading Icon
      - Icon
      - Checklist
      - Avatar
   - Trailing Action (Explanation)
      - Icon
      - Checkbox
      - Radio button
   - Nested
      - Examples of nested list

**Review by Madeline-UX — APPROVED** [review]

**Review by Madeline-UX — APPROVED** [review]

File: changelog.md [hunk 1]
```diff
@@ -1,3 +1,114 @@
...
+export interface ListProps extends svelte.JSX.HTMLAttributes<HTMLUListElement> {
+  disablePadding?: boolean;
+}
+```
+
+### ListItem
+
+- Main component of lists, wrapping text and adornments.
+- Three variants: 'simple' | 'icon' | 'avatar'
+- Three named slots: leadingAdornmnt, trailingAdornment, nested
```
Comment (mike-winberry) [inline changelog.md]: typo: leadingAdornment

File: changelog.md
Comment (mike-winberry) [inline changelog.md:1]: Update with ssx props and box if exists.

File: uui/src/lib/List/List.svelte [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  // Props
+  type $$Props = ListProps;
+  export let disablePadding: boolean | undefined = undefined;
+</script>
+
+<Box
+  element="ul"
+  eventComponent={current_component}
+  {...$$restProps}
+  class="mdc-deprecated-list {$$restProps.class || ''} {disablePadding ? 'disabled-padding' : ''}"
```
Comment (mike-winberry) [inline uui/src/lib/List/List.svelte:15]: Lets add "list" as  a class just for continuity. I want to move away from mdc for the users (it should still be there just want to alleviate the need for users to use the mdc classes)

File: uui/src/lib/List/List.svelte [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  ssx={{
```
Comment (mike-winberry) [inline uui/src/lib/List/List.svelte:16]: need to merge user ssx from $$restProps or the users ssx wont be applied.

File: uui/src/lib/List/ListItem.svelte [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    @include list.deprecated-item-secondary-text-ink-color(var(--on-background));
+  }
+
+  .mdc-deprecated-list-item__graphic,
+  .mdc-deprecated-list-item__meta {
+    color: var(--on-background);
+  }
+
+  // Selected State
+
```
Comment (mike-winberry) [inline uui/src/lib/List/ListItem.svelte:92]: delete this empty line

File: uui/src/lib/List/ListItem.svelte
Comment (mike-winberry) [inline uui/src/lib/List/ListItem.svelte:1]: Wrap in box and open to ssx

File: uui/src/lib/List/ListItem.types.d.ts
Comment (mike-winberry) [inline uui/src/lib/List/ListItem.types.d.ts:1]: List item should extend box props.

File: uui/src/lib/List/ListItemCheckbox.svelte
Comment (mike-winberry) [inline uui/src/lib/List/ListItemCheckbox.svelte:1]: wrap in box and boxprops

File: uui/src/lib/List/ListItemCheckbox.types.d.ts
Comment (mike-winberry) [inline uui/src/lib/List/ListItemCheckbox.types.d.ts:1]: Box props

File: uui/src/lib/List/ListItemCheckbox.svelte
Comment (TristanHoladay) [reply uui/src/lib/List/ListItemCheckbox.svelte:1]: Running into style issues with converting checkbox into Box. Will do this within the #13.

File: uui/src/lib/List/ListItemCheckbox.types.d.ts
Comment (TristanHoladay) [reply uui/src/lib/List/ListItemCheckbox.types.d.ts:1]: Will do this update in #13 when ListItemCheckbox is extracted out as a general checkbox.

**Review by mike-winberry — APPROVED** [review]

**Pull request merged**